data-prepper-plugins/zero-buffer/src/main/java/org/opensearch/dataprepper/plugins/buffer/zerobuffer/ZeroBuffer.java

```diff
@@ -0,0 +1,102 @@
+package org.opensearch.dataprepper.plugins.buffer.zerobuffer;
+
+import org.opensearch.dataprepper.core.pipeline.buffer.AbstractZeroBuffer;
+import org.opensearch.dataprepper.metrics.MetricNames;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.CheckpointState;
+import org.opensearch.dataprepper.model.configuration.PipelineDescription;
+import org.opensearch.dataprepper.model.record.Record;
+import io.micrometer.core.instrument.Counter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import lombok.Getter;
+import lombok.AccessLevel;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
+
+@DataPrepperPlugin(name = "zero_buffer", pluginType = Buffer.class)
+public class ZeroBuffer<T extends Record<?>> extends AbstractZeroBuffer<T> {
+    private static final Logger LOG = LoggerFactory.getLogger(ZeroBuffer.class);
+    private static final String PLUGIN_COMPONENT_ID = "ZeroBuffer";
+    private final PluginMetrics pluginMetrics;
+    private final ThreadLocal<Collection<T>> threadLocalStore;
+    @Getter(value = AccessLevel.PACKAGE)
+    private final String pipelineName;
+    @Getter(value = AccessLevel.PACKAGE)
+    private final Counter writeRecordsCounter;
+    @Getter(value = AccessLevel.PACKAGE)
+    private final Counter readRecordsCounter;
+
+    @DataPrepperPluginConstructor
+    public ZeroBuffer(PipelineDescription pipelineDescription) {
+        this.pluginMetrics = PluginMetrics.fromNames(PLUGIN_COMPONENT_ID, pipelineDescription.getPipelineName());
```

Comment:
I think we can now get this from dependency injection. @san81 , is that correct?